src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -0,0 +1,51 @@
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+
+public interface IItemHandler
+{
+    /**
+     * Returns the number of slots available
+     *
+     * @return The number of slots available
+     **/
+    int getSlots();
+
+    /**
+     * Returns the ItemStack in a given slot.
+     * The resultant value's stacksize may be greater than the stacks max size.
+     * <p/>
+     * IMPORTANT: This ItemStack MUST NOT be modified. This method is not for
+     * altering an inventories contents. Any implementers who are able to detect
+     * modification through this method should throw an exception.
+     * <p/>
+     * SERIOUSLY: DO NOT MODIFY THE RETURNED ITEMSTACK
+     *
+     * @param slot Slot to query
+     * @return ItemStack in given slot. May be null.
+     **/
+    ItemStack getStackInSlot(int slot);
+
+    /**
+     * Inserts an ItemStack into the given slot and return the remainder.
+     * Note: This behaviour is subtly different from IFluidHandlers.fill()
+     *
+     * @param slot     Slot to insert into.
+     * @param stack    ItemStack to insert
+     * @param simulate If true, the insertion is only simulated
+     * @return The remaining ItemStack that was not inserted (if the entire stack is accepted, then return null)
+     **/
+    ItemStack insertItem(int slot, ItemStack stack, boolean simulate);
+
+    /**
+     * Extracts an ItemStack from the given slot. The returned value must be null
+     * if nothing is extracted, otherwise it's stack size must not be greater than amount or the
+     * itemstacks getMaxStackSize().
+     *
+     * @param slot     Slot to extract from.
+     * @param amount   Amount to extract (may be greater than the current stacks max limit)
+     * @param simulate If true, the extraction is only simulated
+     * @return ItemStack extracted from the slot, must be null, if nothing can be extracted
```

Comment:
Thats just a NBT limitation, as they use NBT bytes, and it's signed so -128 - 127
At runtime its just a int so it's fine unless you're going through NBT.